R/Lithuania.R

```diff
@@ -0,0 +1,409 @@
+      unassigned <- self$data$raw %>%
+        mutate(national = ifelse(.data$municipality_name == "Lietuva",
+          "national", "municipality"
+        )) %>%
+        group_by(date, .data$national) %>%
+        summarise(across(
+          all_of(sum_cols),
+          ~ sum(.x, na.rm = TRUE)
+        )) %>%
+        mutate(across(
+          all_of(sum_cols),
+          ~ lead(.x, 1) - .x
+        ),
+        municipality_name = "Unknown",
+        ab_prc_day =
+          if_else(
+            .data$ab_tot_day == 0, 0,
+            .data$ab_pos_day / .data$ab_tot_day
+          ),
+        ag_prc_day =
+          if_else(
+            .data$ag_tot_day == 0, 0,
+            .data$ag_pos_day / .data$ag_tot_day
+          ),
+        pcr_prc_day =
+          if_else(
+            .data$pcr_tot_day == 0, 0,
+            .data$pcr_pos_day / .data$pcr_tot_day
+          ),
+        dgn_prc_day =
+          if_else(
+            .data$dgn_tot_day == 0, 0,
+            .data$dgn_pos_day / .data$dgn_tot_day
+          ),
+        map_colors = NA_character_
+        ) %>%
+        filter(.data$national == "municipality") %>%
+        select(-.data$national)
+
+      # Join unknown locations to main dataset
+      osp_data_w_unassigned <- bind_rows(
+        self$data$raw %>% select(-.data$object_id, -.data$municipality_code),
+        unassigned
+      )
+
+      # Exclude national data based on user param (default = FALSE)
+      if (!self$national_data) {
+        osp_data_w_unassigned <-
+          filter(
+            osp_data_w_unassigned,
+            !.data$municipality_name == "Lietuva"
+          )
+      }
+
+      self$data$clean <- osp_data_w_unassigned %>%
+        mutate(
+          date = as_date(date),
+          tested_new = .data$ab_tot_day + .data$ag_tot_day + .data$pcr_tot_day,
+          deaths_new = .data[[death_field]],
+          recovered_total = .data[[recovered_field]]
+        ) %>%
+        rename(
+          cases_new = .data$incidence,
+          cases_total = .data$cumulative_totals,
+          region_level_2 = .data$municipality_name
+        ) %>%
+        left_join(self$data$codes_lookup, by = c("region_level_2")) %>%
+        select(
+          date, region_level_1, region_level_2,
+          cases_new, cases_total, deaths_new,
+          tested_new, recovered_total,
+          everything()
+        )
+      # If we have not been asked to return all the OSP-provided data,
+      # just select the core data sought by get_regional_data
+      # (default is FALSE)
+      if (!self$all_osp_fields) {
+        self$data$clean <- self$data$clean %>%
+          select(
+            date, region_level_1, region_level_2,
+            level_1_region_code, level_2_region_code,
+            cases_new, cases_total, deaths_new, tested_new, recovered_total
+          )
+      }
+      if (self$level == "1") {
+        self$clean_level_1()
+      } else if (self$level == "2") {
+        self$clean_level_2()
```

Comment:
If level 2 data doesn't need cleaning I would just remove this part and the empty functionn `clean_level_2`

Reply:
Done.